mesg-foundation/js-sdk
# Runner authentication

This adds the support of authentication for the runner.
Now each service registers to the `runner.Register` GRPC API and use the issued token to listen to their executions and give them the possibility to submit results and events.

This is not a breaking change for the API of the service but the execution stream will take slightly longer to be established as the authorisation is required.

This also removes the dependency to the `@mesg/api` library so we will be able to safely remove the grpc API in the API library without impacting the service library

## To test

- Checkout the branch `feature/runner-authentication-test`
- Update the path of your service in `packages/runner/src/providers/runtime/index.ts`
- `npm run build`
- Build the engine on the `dev` branch `make docker-dev`
- `packages/cli/bin/run service:dev path-service-in-the-runner --version local`
- You can now process any task on this service that authenticates itself to the engine and run directly on your host, not docker.
- Docker should be fine but because the library is not released yet, docker cannot fetch the library
